### PR TITLE
Support filtering on additional cdf columns

### PR DESCRIPTION
**Pull request by linzhou-db**
Support filtering on additional cdf columns: _commit_version/_commit_timestamp/_change_type

### CODE REVIEW

File: spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala [hunk 1]
```diff
@@ -130,6 +130,7 @@ private[sharing] abstract class RemoteDeltaCDFFileIndexBase(
     auxPartitionSchema: Map[String, DataType] = Map.empty)
     extends RemoteDeltaFileIndexBase(params) {
 
+  // TODO: linzhou, update this
```
Comment (chakankardb) [inline spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala:133]: Consider adding more details on what needs to be updated.

File: spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala [hunk 2]
```diff
@@ -151,10 +152,13 @@ private[sharing] case class RemoteDeltaCDFAddFileIndex(
   override def listFiles(
     partitionFilters: Seq[Expression],
     dataFilters: Seq[Expression]): Seq[PartitionDirectory] = {
+    val updatedFiles = addFiles.map { a =>
+      AddFileForCDF(a.url, a.id, a.getPartitionValuesInDF, a.size, a.version, a.timestamp, a.stats)
```
Comment (chakankardb) [inline spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala:156]: Question: makePartitionDirectories will call getPartitionValuesInDF again? Would that lead to double inclusion of the cdf columns?
Consider adding some comments.

File: spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala (hunk 1 above)
Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala]: This need to be removed.

File: spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala (hunk 2 above)
Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala:158]: It's not double inclusion, and actually to make them consistent. Here, it's applied to each file, makePartitionDirectories is applying the partitions to the directory path.

**Review by chakankardb — APPROVED** [review]

**Pull request merged**